### ISSUE TITLE
cpe_chrome fails on Windows in exciting ways

### ISSUE THREAD

**Issue by natewalck** [facebook/IT-CPE#237]
## Description
cpe_chrome fails in exciting ways when you try to manage a key that isn't in `ENUM_REG_KEYS`. The error it throws is confusing and doesn't reflect that actual issue.

Because a key is used that it doesn't exist, it dumps *all* the cpe_chrome attributes somehow. As a result, it fails on line 53 when it tries to run `.keys` on the attributes, which no long exist because it wasn't expecting that key. (This is my working hypothesis, but not 100% sure).

## Chef Version
14.12.9
## Platform Version
Windows 10

## Replication Case
Try managing the new key `ExternalProtocolDialogShowAlwaysOpenCheckbox`

## Client Output

```
NoMethodError: cpe_chrome[Configure Google Chrome] (cpe_chrome::default line 15) had an error: NoMethodError: undefined method `keys' for nil:NilClass
```

**Comment by svmastersamurai** [comment]
fce08778ca3c560774e0a74bff30f3d1a7bd65c1

**Comment by erikng** [comment]
So that's an interesting workaround but perhaps there is a better solution here? Rather than have to bring in every single key that could possibly ever created into the cookbook, it seems like instead the admin should be able to pass the type registry key itself.

**Comment by svmastersamurai** [comment]
More than likely, most of this code was written years ago and has served our purposes internally. I am more than happy to review pull requests so long as it doesn't break anything for us.

**Comment by erikng** [comment]
That's fair. So you've never had to manage a new key all these years?

FWIW, MacOS resource supports any key.

**Comment by nmcspadden** [comment]
We add the key when we need to manage it. That has, historically, only been a few times in the last few years, so not really worth the effort of completely re-architecting the entire thing to accommodate any arbitrary registry key type.

Unlike macOS, the Windows registry keys need to be specifically and statically typed. If we made it generic such that the admins had to provide their own key names and type names, that means each individual admin whoever used this cookbook would have to do all that work each time. By maintaining an enum of the keys and types, we provide the most common ones already, making it really easy to add new ones without doing a lot of guesswork.

If someone wants to take on the effort of making this generically supported _AND_ still accommodate the existing list, go for it - but it doesn't really seem that worth the effort in the long run unless you are constantly changing keys around (and Google so far hasn't added that much we've cared about).
